File: VesperBoatClubLogBook/app/services/reportService.js
(function () {
    "use strict";

    angular
        .module("app")
        .service("reportService", function ($http) {
            this.getLogBookMileageAndUsageByBoatReport = function (startDate, endDate) {
                return $http.get("api/report/loadlogbookmileageandusagebyboatreport?startDate=" + startDate + "&endDate=" + endDate);
            };

            this.getBoatsCheckedOutReport = function() {
                return $http.get("api/report/LoadBoatsCheckedOutReport");
            };

            this.getMileageLeaderReport = function() {
                return $http.get("api/report/LoadMileageLeaderReport");
            };

            this.getIndividualMileageReport = function(memberId, startDate, endDate) {
                return $http.get("api/report/LoadIndividualMileageReport?memberId=" + memberId + "&startDate=" + startDate + "&endDate=" + endDate);
            };

            this.getClubMileageByMemberReport = function (startDate, endDate) {
                return $http.get("api/report/loadClubMileageByMemberReport?startDate=" + startDate + "&endDate=" + endDate);
            };

            this.getClubLogBookReport = function(startDate, endDate) {
                return $http.get("api/report/LoadClubLogBookReport?startDate=" + startDate + "&endDate=" + endDate);
            };
        });
})();

File: Vesper.Reports.ReportService/ReportService.cs
using System;
using DataAccess;
using Vesper.Reports.Common;
using Vesper.Reports.ReportServiceContract;

namespace Vesper.Reports.ReportService
{
    public class ReportService : IReportService
    {
        private readonly UowHelper<ReportUnitOfWork> _uow;

        public ReportService(string entityConnectionString)
        {
            _uow = new UowHelper<ReportUnitOfWork>(() => new ReportUnitOfWork(entityConnectionString));
        }

        public LogBookMileageAndUsageByBoat GetLogBookMileageAndUsageByBoatReport(DateTime startDate, DateTime endDate) => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetLogBookMileageAndUsageByBoatReport(startDate, endDate);

            return report;
        });

        public BoatsCheckedOutReport GetBoatsCheckedOutReport() => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetBoatsCheckedOutReport();

            return report;
        });

        public MileageLeaderReport GetMileageLeaderReport() => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetMileageLeaderReport();

            return report;
        });

        public IndividualMileageReport GetIndividualMileageReport(int memberId, DateTime startDate, DateTime endDate) => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetIndividualMileageReport(memberId, startDate, endDate);

            return report;
        });

        public ClubMileageByMemberReport GetClubMileageByMemberReport(DateTime startDate, DateTime endDate) => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetClubMileageByMemberReport(startDate, endDate);

            return report;
        });

        public ClubLogBookReport GetClubLogBookReport(DateTime startDate, DateTime endDate) => _uow.Uow(uow =>
        {
            var report = uow.ReportRepository.GetClubLogBookReport(startDate, endDate);

            return report;
        });
    }
}
